executor/write.go

```diff
@@ -1081,20 +1081,27 @@ func (e *InsertExec) initDupOldRowValue(newRows [][]types.Datum) (err error) {
 	for _, keysInRow := range e.uniqueKeysInRows {
 		for _, k := range keysInRow {
 			if val, found := e.dupKeyValues[string(k.key)]; found {
-				var handle int64
-				handle, err = decodeOldHandle(k, val)
-				if err != nil {
-					return errors.Trace(err)
+				if k.isRecordKey {
+					e.dupOldRowValues[string(k.key)] = val
```

Comment:
extract `string(k.key)` as `key = string(k.key)`

Reply:
It is only a cast for the map key. I think it is not necessary.

Reply:
@XuHuaiyu 
If we do this, we need to pay the cost of memory allocation.
Go has an optimization for this.